### ISSUE TITLE
@run is impure by default

### ISSUE THREAD

**Issue by ehird** [lambdabot/lambdabot#58]
`Lambdabot.Plugin.Haskell.Eval.Trusted` re-exports `Lambdabot.Plugin.Haskell.Check.ShowQ`, which exports `myquickcheck`, a non-deterministic (and hence impure) function, since QuickCheck relies on random generation of input parameters. This seems undesirable; QuickCheck is already available through the `@check` command, anyway. Perhaps the import/re-export should simply be removed?

It also re-exports `Math.OEIS`, which is even more blatantly impure (it does network connections!), though the `@oeis` interface is considerably more restricted, so there's not as clear a case for removal... still, I think `@run` should be as pure as possible.

(Also, the `Show` instance for `Mu` is missing; is this because it requires `UndecidableInstances`?)

**Comment by int-e** [comment]
There used to be a serious security hole lurking here: QuickCheck's `Prop` type allows embedding of IO actions, which `@check` would run. I switched to [QuickCheck-safe](https://hackage.haskell.org/package/QuickCheck-safe) some years ago to plug that one. (see 7706db6eab3720fbff374d88baa3c941f1d0ab84)

There's still impurity, because `myquickcheck` embeds a random number generator as a pure function, but it's less dramatic than before.

On the lambdabot instance running on Freenode, the OEIS module is also gone, but mainly because I sandboxed evaluation with no network access.
